_posts/2024-10-18-Mastering-Waveform-Operations-using-Subnero-Modems.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+comments: true
+title: Mastering Waveform Operations using Subnero Modems.md
+date: 18/10/2024
+author: Manu Ignatius
+categories: howto
+feature-img: "assets/img/modems-2024.jpg"
+thumbnail: "assets/img/modems-2024.jpg"
+tags: [howto, modems, baseband]
+---
+
+In underwater communication, where environmental factors such as noise, multipath, Doppler etc. significantly influence performance, it becomes crucial to understand the channel conditions. To gain insights, users often need to record the channel and transmit pilot signals that are not typical communication signals. Additionally, some users may have custom communication signals different from those provided by vendors, requiring flexibility in transmission. This need is prevalent in fields like scientific research and defense, where the ability to transmit and receive arbitrary waveforms within the available frequency band is essential. Subnero modems address these needs by offering multiple methods for both recording and transmitting user-defined signals, providing versatile solutions for various use cases. This article explores the different approaches Subnero modems support for recording and transmitting arbitrary waveforms.
+
+## Recording Incoming Waveforms
+There are multiple methods for capturing incoming waveforms, each catering to specific use cases:
+
+### 1. Recording a Set Number of Samples
+This method is ideal when you need to capture a specific number of samples. You can trigger the recording immediately or schedule it for a specific time using the `bbrec` command. For example, `bbrec 1000` will record 1000 baseband samples. The samples recorded are in the baseband format.
+
+Subnero modems utilize the UnetStack framework, which employs internal messages for communication among different UnetStack agents. The bbrec command internally utilizes the `RecordBasebandSignalReq` to record the samples. `bbrec 1000` is analogous to `bb << new RecordBasebandSignalReq(1000)`.
```

Comment:
This comment reeks of ChatGPT / GenAI.

Reply:
Rephrased both the paragraphs to update the wordings